src/app/drive/components/FileViewer/FileViewer.tsx

```diff
@@ -42,18 +49,72 @@ const FileViewer = ({ file, onClose, onDownload, downloader, show }: FileViewerP
 
   const [blob, setBlob] = useState<Blob | null>(null);
 
+  const dispatch = useAppDispatch();
+  const isTeam = useAppSelector(sessionSelectors.isTeam);
+  const userEmail: string = localStorageService.getUser()?.email || '';
+
   useEffect(() => {
     if (isTypeAllowed && show) {
       const abortController = new AbortController();
 
-      downloader(abortController).then(setBlob).catch((err) =>{
-        if (abortController.signal.aborted) {
-          return;
-        } 
-        
-        console.error(err);
-      });
-
+      downloader(abortController)
+        .then(async (fileBlob) => {
+          setBlob(fileBlob);
+          if (file) {
+            const currentThumbnail = file.thumbnails && file.thumbnails.length > 0 ? file.thumbnails[0] : null;
+            const fileObject = new File([fileBlob], file.name);
+            const fileUpload: FileToUpload = {
+              name: file.name,
+              size: file.size,
+              type: file.type,
+              content: fileObject,
+              parentFolderId: file.folderId,
+            };
+
+            const thumbnail = await getThumbnailFrom(fileUpload);
+
+            //Don't upload generated thumbnail if it match in size and type with currentThumbnail
+            if (thumbnail && thumbnail.file && thumbnail.type &&
+              (!currentThumbnail ||
+                (currentThumbnail && (Number(currentThumbnail.size) !== Number(thumbnail.file.size)
+                  || String(currentThumbnail.type) !== String(thumbnail.type)
+                  || (Number(currentThumbnail.max_width)) !== (Number(thumbnail.max_width))
+                  || (Number(currentThumbnail.max_height)) !== (Number(thumbnail.max_height)))))) {
```

Comment:
This `if` looks a bit complicated and difficult to read, perhaps creating a function called `compareThumbnails` is more readable, and then you can remove the comment as it is obvious what it is doing